libyul/backends/wasm/EVMToEWasmTranslator.cpp

```diff
@@ -371,8 +808,8 @@ function callvalue() -> z1, z2, z3, z4 {
 	z1, z2, z3, z4 := mload_internal(0)
 }
 function calldataload(x1, x2, x3, x4) -> z1, z2, z3, z4 {
-	eth.callDataCopy(0, u256_to_i32(x1, x2, x3, x4), 32)
-	z1, z2, z3, z4 := mload_internal(0)
+	z1, z2, z3, z4 := calldatasize()
```

Comment:
Can you explain why you needed to change this?

Reply:
I assumed that calldataload should load all data starting from position (x1, x2, x3, x4). In previous implementation it copied only 32 bytes starting from (x1, x2, x3, x4). But I checked evm documentation again and it this instruction copies always 256 bits from msg.data starting from  (x1, x2, x3, x4). My bad. Sorry should stay as it was before.